code/modules/power/solar.dm

```diff
@@ -47,7 +48,15 @@
 		id = "zeta"
 	aisat
 		id = "aisat"
-
+	New()
+		..()
+		for(var/obj/machinery/power/data_terminal/test_link in powernet.data_nodes) // plug and play
+			if(!istype(test_link?.master,/obj/machinery/computer/solar_control)) continue
+			var/obj/machinery/computer/solar_control/S = test_link?.master
```

Comment:
one letter var names bad (especially since S is used for several different types here)